torch/csrc/jit/pybind_utils.h

```diff
@@ -174,25 +232,53 @@ inline TypedIValue toTypedIValue(py::handle input) {
   }
 }
 
+inline IValue toIValue(
+    py::handle obj,
+    const TypePtr& type,
+    c10::optional<int32_t> N = c10::nullopt);
+
+inline TypedIValue toTraceableIValue(py::handle input) {
+  try {
+
+  auto match = tryToInferType(input);
+  if (!match.type) {
+    AT_ERROR(
+        "Tracer cannot infer type of ", py::str(input), "\n:", match.errMsg);
+  }
+  auto type = *match.type;
+
+  if (type->kind() == TypeKind::ListType ||
+      type->kind() == TypeKind::DictType ||
+      type->kind() == TypeKind::TupleType ||
+      type->isSubtypeOf(TensorType::get())) {
+    return TypedIValue(toIValue(input, type), type);
```

Comment:
I think atm we only support Lists / Dicts / Tuples of Tensors right ? or nested ones ? i'm not entirely sure. Don't think you're checking for that case.